### PR TITLE
fix: [PIE-7681]: pipeline chaining support for input set & triggers with multiple entities

### PR DESCRIPTION
**Pull request by Eric761**
### Summary

Added custom hook `useGetResolvedChildPipeline` to merge `resolvedPipeline` of parent with all child Pipelines, returning a `loader` and `resolvedMergedPipeline`

#### Screenshots

Screenshot 2023-01-18 at 8 19 15 AM

Screenshot 2023-01-18 at 8 18 59 AM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by vkbansal-harness — DISMISSED** [review]

File: src/modules/70-pipeline/components/PipelineDiagram/Nodes/StepGroupNode/StepGroupNode.tsx [hunk 1]
```diff
@@ -267,7 +267,6 @@ export function StepGroupNode(props: any): JSX.Element {
                     e.stopPropagation()
                   }}
                   flex={{ alignItems: 'center', justifyContent: 'flex-start' }}
-                  width={'100%'}
```
Comment (rameshx) [inline src/modules/70-pipeline/components/PipelineDiagram/Nodes/StepGroupNode/StepGroupNode.tsx:270]: This is added to make the step group name get ellipsis on overflow. Can you check if it's working?

**Review by pankajsingh-harness — CHANGES_REQUESTED** [review]

**Review by satyam-harness — APPROVED** [review]

File: src/modules/72-triggers/components/Triggers/WebhookTrigger/WebhookTriggerWizard.tsx [hunk 2]
```diff
@@ -130,6 +131,7 @@ export default function WebhookTriggerWizard(
 
   const [yamlHandler, setYamlHandler] = useState<YamlBuilderHandlerBinding | undefined>()
   const [selectedView, setSelectedView] = useState<SelectedView>(SelectedView.VISUAL)
+  const [_resolvedPipeline, setResolvedPipeline] = useState<PipelineInfoConfig | undefined>()
```
Comment (chetanvekariya-harness) [inline src/modules/72-triggers/components/Triggers/WebhookTrigger/WebhookTriggerWizard.tsx:134]: `_`

File: src/modules/70-pipeline/hooks/useGetResolvedChildPipeline.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+              resolvedChildPipelines.set(key, resolvedChildPipeline)
+            }
+          }
+        })
+        if (!isEmpty(resolvedChildPipelines)) {
+          setResolvedChildPipelineMap(resolvedChildPipelines)
+        }
+        setLoadingResolvedChildPipeline(false)
+      })
+      .catch(_e => {
```
[the remark this answers is not in the document]
Comment (Eric761) [reply src/modules/70-pipeline/hooks/useGetResolvedChildPipeline.ts:73]: done

File: src/modules/70-pipeline/hooks/useGetResolvedChildPipeline.ts [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+import { yamlParse } from '@common/utils/YamlHelperMethods'
+import { StageType } from '@pipeline/utils/stageHelpers'
+
+export interface UseGetResolvedChildPipelineReturn {
+  loadingResolvedChildPipeline: boolean
+  resolvedMergedPipeline?: PipelineInfoConfig
+}
+
+const getChildPipelineKey = (data: StageElementWrapperConfig): string => {
+  const childPipelineId = get(data, 'stage.spec.pipeline')
```
[the remark this answers is not in the document]
Comment (Eric761) [reply src/modules/70-pipeline/hooks/useGetResolvedChildPipeline.ts:28]: updated interface

File: src/modules/70-pipeline/hooks/useGetResolvedChildPipeline.ts [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  const childPipelineProject = get(data, 'stage.spec.project')
+  const childPipelineOrg = get(data, 'stage.spec.org')
```
[the remark this answers is not in the document]
Comment (Eric761) [reply src/modules/70-pipeline/hooks/useGetResolvedChildPipeline.ts:30]: done

File: src/modules/70-pipeline/components/PipelineDiagram/Nodes/StepGroupNode/StepGroupNode.tsx (hunk 1 above)
Comment (Eric761) [reply src/modules/70-pipeline/components/PipelineDiagram/Nodes/StepGroupNode/StepGroupNode.tsx:270]: Made the changes, will work for both normal and pipeline chaining case.

**Review by rameshx — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by vkbansal-harness — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Pull request merged**